models/issues/milestone.go

```diff
@@ -361,7 +362,11 @@ func (opts GetMilestonesOption) toCond() builder.Cond {
 	}
 
 	if len(opts.Name) != 0 {
-		cond = cond.And(builder.Like{"UPPER(name)", strings.ToUpper(opts.Name)})
+		if setting.Database.UseSQLite3 {
+			cond = cond.And(builder.Like{"UPPER(name)", util.ToUpperASCII(opts.Name)})
```

Comment:
I think a comment would be good explaining basically what you wrote in the description.
Otherwise it isn't immediately clear why this is done.

Reply:
+1. How about introducing a common function like `db.BuildLikeUpper(key, value)`

```go
func BuildLikeUpper(key, value) {
    if setting.Database.UseSQLite3 {
        // detailed comments ..........
        return builder.Like{"UPPER(...)", util.ToUpperASCII(value)}
    } else {
        return builder.Like{"UPPER(...)", strings.ToUpper(value)}
    }
}
```

Then the code from https://github.com/go-gitea/gitea/pull/18437  can also use this common function